net/src/headers.rs

```diff
@@ -471,7 +472,10 @@ impl Headers {
                         trace!("no transport header: can't update checksum")
                     }
                     Some(transport) => {
-                        transport.update_checksum(net, payload.as_ref());
+                        if !is_vxlan {
+                            // only recompute checksum if it is not vxlan
+                            transport.update_checksum(net, payload.as_ref());
```

Comment:
This is better and okay for now.  Will cause problems if the checksum comes in non-zero and we manipulate the payload without reencap.  Pretty niche case, so no need to worry for the foreseeable future.